python_modules/dagster/dagster/_core/storage/asset_check_execution_record.py

```diff
@@ -24,12 +25,36 @@ class AssetCheckExecutionResolvedStatus(enum.Enum):
     SKIPPED = "SKIPPED"  # the run finished, didn't fail, but the check didn't execute
 
 
-class AssetCheckExecutionRecord(NamedTuple):
-    id: int
-    run_id: str
-    status: AssetCheckExecutionRecordStatus
-    evaluation_event: Optional[EventLogEntry]
-    create_timestamp: float
+class AssetCheckExecutionRecord(
```

Comment:
Would be a great use-case for pydantic since we're not serializing here. Also would involve less boilerplate too. Can check out https://github.com/dagster-io/internal/pull/5458 as a reference. Up to you.

Reply:
Fancy, I forget about those. Could always refactor after the rush